package.json

```diff
@@ -7,33 +7,24 @@
     "dev": "vitepress",
     "build": "vitepress build",
     "preview": "vitepress preview",
-    "preinstall": "npx only-allow pnpm"
+    "preinstall": "npx only-allow pnpm",
+    "type": "vue-tsc --noEmit"
   },
   "dependencies": {
-    "@vue/repl": "^4.0.1",
-    "@vue/theme": "^2.2.5",
+    "@vue/repl": "^4.1.2",
+    "@vue/theme": "^2.2.8",
     "dynamics.js": "^1.1.5",
-    "gsap": "^3.9.0",
-    "vitepress": "^1.1.0",
-    "vue": "^3.4.0"
+    "gsap": "^3.12.5",
+    "vitepress": "^1.2.0",
+    "vue": "^3.4.27"
   },
   "devDependencies": {
-    "@types/markdown-it": "^12.2.3",
-    "@types/node": "^20.10.1",
-    "terser": "^5.14.2"
+    "@types/body-scroll-lock": "^3.1.2",
+    "@types/markdown-it": "^14.1.1",
+    "@types/node": "^20.12.12",
+    "terser": "^5.31.0",
+    "typescript": "^5.4.5",
+    "vue-tsc": "^2.0.19"
   },
-  "pnpm": {
-    "overrides": {
-      "@vitejs/plugin-vue": "5.0.0-beta.1"
-    },
-    "peerDependencyRules": {
-      "ignoreMissing": [
-        "@algolia/client-search",
-        "react",
-        "react-dom",
-        "@types/react",
-        "search-insights"
-      ]
-    }
-  }
+  "packageManager": "pnpm@9.1.1"
```

Comment:
@brc-dd do you think we can skim this rule? Since I guess it breaks the translation deployments on Vercel like docs-cs https://github.com/vuejs-translations/docs-cs/issues/132#issuecomment-2122103463

Reply:
We can fix it on v8 for now. Vercel supports corepack but I think they are blocked on something in pnpm v9. On newer node versions, this is automatically added each time you run a command and have corepack enabled locally (say, for other projects). Also, the engine fields of most of the translation repos are wrong. vite, vitepress, pnpm don't support node v14.

Reply:
Or I think v9 is also supported via corepack (but detecting it is not supported). Try setting https://vercel.com/docs/deployments/configure-a-build#corepack `ENABLE_EXPERIMENTAL_COREPACK=1` in env variable.